pkg/kubelet/container_bridge.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"bytes"
+	"net"
+	"os/exec"
+	"regexp"
+
+	"github.com/golang/glog"
+)
+
+var cidrRegexp = regexp.MustCompile(`inet ([0-9a-fA-F.:]*/[0-9]*)`)
+
+func ensureCbr0(wantCIDR *net.IPNet) error {
+	if !cbr0CidrCorrect(wantCIDR) {
+		glog.V(2).Infof("Attempting to recreate cbr0 with address range: %s", wantCIDR)
+
+		// delete cbr0
+		if err := exec.Command("ip", "link", "set", "dev", "cbr0", "down").Run(); err != nil {
+			glog.Error(err)
+			return err
+		}
+		if err := exec.Command("brctl", "delbr", "cbr0").Run(); err != nil {
+			glog.Error(err)
+			return err
+		}
+		// recreate cbr0 with wantCIDR
+		if err := exec.Command("brctl", "addbr", "cbr0").Run(); err != nil {
+			glog.Error(err)
+			return err
+		}
+		if err := exec.Command("ip", "addr", "add", wantCIDR.String(), "dev", "cbr0").Run(); err != nil {
+			glog.Error(err)
+			return err
+		}
+		if err := exec.Command("ip", "link", "set", "dev", "cbr0", "up").Run(); err != nil {
+			glog.Error(err)
+			return err
+		}
+		// restart docker
+		if err := exec.Command("service", "docker", "restart").Run(); err != nil {
```

Comment:
Here is the doc for coreos and rkt's network configure: 

https://github.com/coreos/rkt/blob/master/Documentation/networking.md#bridge

We don't need cbr0 here. This is why I suggested not to return error here instead of logging. Also please add a TODO(dawnchen): figure out how to handle for rkt case here? I might need to do a refactory later to push this down to container_runtime (docker, rkt) library.